omod/src/main/java/org/openmrs/module/idgen/rest/search/LogEntrySearchHandler.java

```diff
@@ -0,0 +1,75 @@
+import org.openmrs.module.webservices.rest.web.resource.api.SearchConfig;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchHandler;
+import org.openmrs.User;
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchQuery;
+import org.openmrs.module.webservices.rest.web.resource.impl.EmptySearchResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.springframework.stereotype.Component;
+
+@Component
+public class LogEntrySearchHandler implements SearchHandler {
+
+    private final SearchConfig searchConfig = new SearchConfig("default", RestConstants.VERSION_1 + "/logentry",
+            Arrays.asList("1.8.*", "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*"),
+            Arrays.asList(new SearchQuery.Builder(
+                    "Allows you to find log of ID Generation Activities by Source Name, Identifier contents,Generated Date Range,Comment contents and User who generated the log entry")
+                            .withOptionalParameters("source", "identifier", "fromDate", "toDate", "comment",
+                                    "generatedBy")
+                            .build()));
+
+    @Override
+    public PageableResult search(RequestContext context) throws ResponseException {
+        IdentifierSourceService identifierSourceService = Context.getService(IdentifierSourceService.class);
+        UserService service = Context.getUserService();
+
+        String source = context.getRequest().getParameter("source");
+        String fromDate = context.getRequest().getParameter("fromDate");
+        String toDate = context.getRequest().getParameter("toDate");
+        String identifier = context.getRequest().getParameter("identifier");
+        String comment = context.getRequest().getParameter("comment");
+        String generatedBy = context.getRequest().getParameter("generatedBy");
+
+        List<LogEntry> logEntries = new ArrayList<LogEntry>();
+
+        IdentifierSource logSource = source != null ? identifierSourceService.getIdentifierSourceByUuid(source) : null;
+        Date dateFrom = fromDate != null ? (Date) ConversionUtil.convert(fromDate, Date.class) : null;
+        Date dateTo = toDate != null ? (Date) ConversionUtil.convert(toDate, Date.class) : null;
+        User user = generatedBy != null ? service.getUserByUuid(generatedBy): null;
+        if (logSource == null && identifier == null && comment == null && dateFrom == null && dateTo == null
+                && user == null) {
+            return new EmptySearchResult();
+        }
+        logEntries = identifierSourceService.getLogEntries(logSource, dateFrom, dateTo, identifier, user, comment);
+
+        return new NeedsPaging<LogEntry>(logEntries, context);
+    }
+    public SearchConfig getSearchConfig() {
```

Comment:
add `@override` statement so that your search can be used as the search configuration.

Reply:
Its a shame I'd missed that. Fixed it too. Thanks